src/manifest.c

```diff
@@ -417,6 +417,9 @@ gboolean save_manifest_file(const gchar *filename, RaucManifest *mf, GError **er
 
 		group = g_strconcat(RAUC_IMAGE_PREFIX ".", image->slotclass, NULL);
 
+		if (image->variant)
+			 group = g_strconcat(group, ".", image->variant, NULL);
```

Comment:
`g_strconcat()` allocates a new string but we have to free the old one before.

Something like 

    if (image->variant) {
       gchar *tmp = group;
       group = g_strconcat(group, ...)
       g_free(tmp);
    }

should do it.